ext/flash/sendfile.rs

```diff
@@ -10,20 +10,19 @@ use std::task::{self, Poll};
 pub struct SendFile {
   pub io: (RawFd, RawFd),
   pub written: usize,
+  pub count: usize,
 }
 
 impl SendFile {
   #[inline]
   pub fn try_send(&mut self) -> Result<usize, std::io::Error> {
     #[cfg(target_os = "linux")]
     {
-      // This is the maximum the Linux kernel will write in a single call.
-      let count = 0x7ffff000;
```

Comment:
This should probably stay... or at least, some of the time. Buckle up, it's going to get complicated.

1. In non-largefile mode (really only relevant to old 32 bits systems), the limit is INT_MAX-1 (2**31-1)
2. In largefile mode, the limit is either 2**48 or 2**63
3. Except when the file system imposes a lower limit

There's no real rhyme or reason to (3) but 2 GB is a safe lower bound. Grep the kernel's fs/ directory for `s_maxbytes` to get an idea of the wildly varying maximum sizes.

<hr>

Apropos EOVERFLOW: the kernel can return that error even after writing data. You need to check if `offset` changed to figure out whether it did.